src/types/tcb_info.rs

```diff
@@ -0,0 +1,399 @@
+use anyhow::{bail, Context};
+use chrono::Utc;
+use p256::ecdsa::signature::Verifier;
+use p256::ecdsa::VerifyingKey;
+use serde::{Deserialize, Serialize};
+use serde_json::value::RawValue;
+
+use super::{report::TdxReportBody, sgx_x509::SgxPckExtension};
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct TcbInfoAndSignature {
+    #[serde(rename = "tcbInfo")]
+    tcb_info_raw: Box<RawValue>,
+    #[serde(with = "hex")]
+    signature: Vec<u8>,
+}
+
+impl TryFrom<String> for TcbInfoAndSignature {
+    type Error = serde_json::Error;
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        serde_json::from_str(&value)
+    }
+}
+
+impl TcbInfoAndSignature {
+    pub fn as_tcb_info_and_verify(&self, public_key: VerifyingKey) -> anyhow::Result<TcbInfo> {
```

Comment:
`tcb_info` expiration check is missing.

Reply:
Done here https://github.com/automata-network/dcap-rs/pull/15/commits/3bb57c59fa8dd2c1d017ad8d76452556db96a4dd